src/GafferTestModule/SignalsTest.cpp

```diff
@@ -94,12 +94,20 @@ void testDisconnectMatchingLambda()
 	auto connection2 = signal.connect( slot2 );
 	GAFFERTEST_ASSERTEQUAL( signal.numSlots(), 2 );
 
+#ifdef _MSC_VER
+	signal.disconnect( static_cast<void( __cdecl * )()>( slot1 ) );
+#else
```

Comment:
Ugh! I'm way out of my depth here, but it would be great if we could fix this on the Signals side, rather than require client code to do the weird cast. I'm pretty sure this test was OK with `boost::signals`, so presumably there's some voodoo we could put inside `disconnect()` to make this possible?

Reply:
Scratch that - this one is definitely my problem and not yours. Could you just add a todo saying "Can this be dealt with internally in `Signal::disconnect()?"?

Reply:
Of course. If it's not too much of a distraction, what led you to that conclusion? This one has made it into my top five "why can't Windows just cooperate" list and I don't 100% understand why it can't do the comparison itself.

Reply:
> what led you to that conclusion?

It's certainly not based on any belief that I know enough to fix it! I just figured that the ideal fix is inside `Signal` and that your mission to get Gaffer on Windows didn't really need to get held up by it...

Reply:
I added the todo in 754fe935bdee94c056df89cefadbbfc7044ab130